build.gradle

```diff
@@ -203,6 +203,10 @@ project(':iceberg-common') {
 }
 
 project(':iceberg-core') {
+  test {
+    useJUnitPlatform()
+  }
```

Comment:
We have a lot of parameterized tests that don't require JUnit5. Here's an example: https://github.com/apache/iceberg/blob/master/api%2Fsrc%2Ftest%2Fjava%2Forg%2Fapache%2Ficeberg%2Fexpressions%2FTestExpressionSerialization.java#L31-L72

I generally prefer not to change the build in small PRs unless we need to. If we want to move to JUnit5, then we should take a look at that and discuss it on the dev list. My main concern there is the reordering of assert method arguments.